SwissTransferFeatures/SettingsView/EditSettings/EditSettingsDatasource.swift

```diff
@@ -0,0 +1,189 @@
+/*
+ Infomaniak SwissTransfer - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakDI
+import STCore
+import STResources
+import SwiftUI
+import SwissTransferCore
+import SwissTransferCoreUI
+
+/// Something used to populate a `EditSettingsView`
+protocol EditCellModel {
+    var label: String { get }
+    var action: () -> Void { get }
+    var leftIconAsset: STResourcesImages? { get }
+}
+
+struct EditCellDatom: EditCellModel, Hashable, Equatable {
+    let label: String
+    let action: () -> Void
+    let leftIconAsset: STResourcesImages?
+    let rightIconAsset: STResourcesImages?
+
+    init(label: String,
+         action: @escaping () -> Void,
+         leftIconAsset: STResourcesImages? = nil,
+         rightIconAsset: STResourcesImages? = nil) {
+        self.label = label
+        self.action = action
+        self.leftIconAsset = leftIconAsset
+        self.rightIconAsset = rightIconAsset
+    }
+
+    static func == (lhs: Self, rhs: Self) -> Bool {
+        lhs.label == rhs.label
+    }
+
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(label)
+    }
+}
+
+protocol EditSettingsModel {
+    var source: SettingDetailUi { get }
+    var title: String { get }
+    var cellsModel: [EditCellDatom] { get }
+}
+
+struct EditThemeDatasource: EditSettingsModel {
+    let source = SettingDetailUi.theme
+    let title = STResourcesStrings.Localizable.settingsThemeTitle
+    let selectedTheme: Theme?
+    var cellsModel = [EditCellDatom]()
+
+    init(appSettings: AppSettings?) {
+        selectedTheme = appSettings?.theme
+        cellsModel = Theme.allCases.map { setting in
+            EditCellDatom(
+                label: setting.localized,
+                action: action(forSetting: setting),
+                rightIconAsset: rightIconAsset(forSetting: setting)
+            )
+        }
+    }
+
+    private func rightIconAsset(forSetting setting: Theme) -> STResourcesImages? {
```

Comment:
I mean I can factorise this bit with some generics but at the cost of more complexity that will be hard to work with for everyone. So not doing it.

Reply:
@PhilippeWeidmann I had a chat with @valentinperignon, we defined what should change, and I'll use some generic data structure to be able to have a more abstract navigation.

Reply:
Discussed I should decorate STCore types instead